classes/Context.php

```diff
@@ -450,18 +450,20 @@ public function getTranslatorFromLocale($locale)
         $withDB = !$this->language instanceof InstallLanguage;
         $theme = $this->shop !== null ? $this->shop->theme : null;
 
-        try {
-            $containerFinder = new ContainerFinder($this);
-            $containerFinder->getContainer()->get('prestashop.translation.translator_language_loader')
-                ->setIsAdminContext($adminContext)
-                ->loadLanguage($translator, $locale, $withDB, $theme);
-        } catch (ContainerNotFoundException $exception) {
-            // If a container is still not found, instantiate manually the translator loader
-            // This will happen in the Front as we have legacy controllers, the Sf container won't be available.
-            // As we get the translator in the controller's constructor and the container is built in the init method, we won't find it here
-            (new TranslatorLanguageLoader(new ModuleRepository()))
-                ->setIsAdminContext($adminContext)
-                ->loadLanguage($translator, $locale, $withDB, $theme);
+        if ($this instanceof Context) {
```

Comment:
This one is a bit weird too.

Is there a standard usecase where `$this` would not be an instance of `Context`?

If no, I'm worried that we're adding code "just to please phpstan" and this is not an improvement.

Reply:
:warning: ContainerFinder has a parameter `Context` and not `ContextCore`, so we must check if it's good.

Or I break the constructor of `ContainerFinder`.

Reply:
@atomiix `ContainerFinder` wait for a `Context` class and not a `ContextCore` class.

PHPStan doesn't know rules for PrestaShop Extends. It knows PHP inheritance.

So a class `TrucMuche` can inherit from `ContextCore` and could be injected to `ContainerFinder`.

So it's why I check this. Or I break the constructor of ContainerFinder to define the parameter to `ContextCore`.

Reply:
But `Context` inherits from `ContextCore`. I think it's only a matter of configuration. Maybe the class_stub.php should be included in a way.

Reply:
`Context` is parent of `ContextCore`

`$this` is `ContextCore` or `Context` or `class parent of ContextCore` (`TrucMuche` in my previous sample).

`ContainerFinder` wants `Context` : so we must be sure to have a `Context`.

Reply:
I don't think so, `Context` is **child** of `ContextCore`:
<img width="310" alt="Capture d’écran 2022-02-15 à 10 55 04" src="https://user-images.githubusercontent.com/2168836/154037804-8a284678-40e9-4808-bb79-79154f76c0bc.png">
Or I'm missing something 🤨